uber/site_sections/preregistration.py

```diff
@@ -162,7 +162,12 @@ def form(self, session, message='', edit_id=None, **params):
                 else:
                     target = group if group.badges else attendee
                     track_type = c.EDITED_PREREG if target.id in self.unpaid_preregs else c.UNPAID_PREREG
-                    self.unpaid_preregs[target.id] = to_sessionized(attendee, group)
+                    if attendee.paid == c.NEED_NOT_PAY:
```

Comment:
Hm, I didn't realize it at the time but this is actually something we probably want to revert. Its effect is that if you use a promo code that makes your badge free, you get taken right to the confirmation page -- but we already have similar functionality in the index page, which calls `process_free_prereg` if the total $ in your cart is <= 0.